zipline/gens/tradesimulation.py

```diff
@@ -272,14 +272,20 @@ def past_auto_close_date(asset):
         for asset in assets_to_clear:
             perf_tracker.process_close_position(asset, dt, data_portal)
 
-        # Remove open orders for any sids that have reached their
-        # auto_close_date.
+        # Remove open orders for any sids that have reached their auto close
+        # date. These orders get processed immediately because otherwise they
+        # would not be processed until the first bar of the next day.
         blotter = algo.blotter
-        assets_to_cancel = \
-            set([asset for asset in blotter.open_orders
-                 if past_auto_close_date(asset)])
+        assets_to_cancel = [
+            asset for asset in blotter.open_orders
+            if past_auto_close_date(asset)
+        ]
```

Comment:
No reason pertinent to this PR. But `blotter.open_orders` is just a dictionary keyed by assets, which means they're already unique, so I didn't see a need for this to be a set. And because all we do with it later is just iterate over it.